frame/election-provider-support/src/onchain.rs

```diff
@@ -85,13 +86,9 @@ pub trait Config {
 	/// always be more than `DataProvider::desired_target`.
 	type MaxWinners: Get<u32>;
 
-	/// Bounds the number of voters, when calling into [`Config::DataProvider`]. It might be
-	/// overwritten in the `InstantElectionProvider` impl.
-	type VotersBound: Get<u32>;
-
-	/// Bounds the number of targets, when calling into [`Config::DataProvider`]. It might be
-	/// overwritten in the `InstantElectionProvider` impl.
-	type TargetsBound: Get<u32>;
+	/// Elections bounds, to use when calling into
+	/// [`Config::DataProvider`]. It might be overwritten in the `InstantElectionProvider` impl.
+	type ElectionBounds: Get<ElectionBounds>;
```

Comment:
```suggestion
	type Bounds: Get<ElectionBounds>;
```

Because, well, we are in a crate who's name starts with election 😅

Or `DataProviderBounds` is more accurate, implying the this is how much it fetches from the data provider.